packages/flutter_tools/test/integration.shard/android_plugin_compilesdkversion_mismatch_test.dart

```diff
@@ -0,0 +1,124 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:file_testing/file_testing.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/io.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/dart/pub.dart';
+import 'package:flutter_tools/src/reporting/reporting.dart';
+
+import '../src/common.dart';
+import '../src/fakes.dart';
+import 'test_utils.dart';
+
+void main() {
+
+  Directory tempDir;
+
+  setUp(()  {
+    Cache.flutterRoot = getFlutterRoot();
+    tempDir = createResolvedTempDirectorySync('flutter_plugin_test.');
+  });
+
+  tearDown(() async {
+    tryToDelete(tempDir);
+  });
+
+  test('error logged when plugin Android compileSdkVersion higher than project', () async {
+    final String flutterBin = fileSystem.path.join(
+      getFlutterRoot(),
+      'bin',
+      'flutter',
+    );
+
+    final RegExp androidCompileSdkVersionRegExp = RegExp(r'compileSdkVersion ([0-9]+|flutter.compileSdkVersion)');
+
+    // Create dummy plugin
+    processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'create',
+      '--template=plugin',
+      '--platforms=android',
+      'test_plugin',
+    ], workingDirectory: tempDir.path);
+
+    final Directory pluginAppDir = tempDir.childDirectory('test_plugin');
+    final File pluginGradleFile = pluginAppDir.childDirectory('android').childFile('build.gradle');
+    expect(pluginGradleFile, exists);
+    final String pluginBuildGradle = pluginGradleFile.readAsStringSync();
+
+
+    // Bump up plugin compileSdkVersion to 31
+    final String newPluginGradleFile = pluginBuildGradle.replaceAll(
+      androidCompileSdkVersionRegExp, 'compileSdkVersion 31');
+    pluginGradleFile.writeAsStringSync(newPluginGradleFile);
+
+    // Create dummy project
+    processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'create',
+      '--platforms=android',
+      'test_project',
+    ], workingDirectory: tempDir.path);
```

Comment:
it could run `flutter build apk` in the plugin's example app directory, so I don't think it would need another app project.